dao/users.js

```diff
@@ -0,0 +1,108 @@
+const debug = require('debug')('calendar-server:reminders');
+const bcrypt = require('bcrypt');
+
+const database = require('./database');
+const { NotFoundError } = require('../utils/errors');
+const {
+  checkPropertyType, checkIsArray
+} = require('../utils/object_validator.js');
+
+function notFoundError(id) {
+  return NotFoundError.createWithSubject('user', { name: 'id', value: id });
+}
+
+const BCRYPT_SALT_ROUNDS = 10;
+const SHA_HASH_SALT = 'todochangeme';
+
+function bcryptHash(plainText) {
+  return new Promise((resolve, reject) => {
+    bcrypt.hash(plainText, BCRYPT_SALT_ROUNDS, function(err, hash) {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      resolve(hash);
+    });
+  });
+}
```

Comment:
note: some "best practices" pages talk about using base64-encoded SHA-384 hash with bcrypt because of its limitations (72 first characters for example). Or we could use the newest argon2 algorithm directly, as there are wrappers for node.